va/va_vpp.h

```diff
@@ -286,22 +286,152 @@ typedef enum _VAProcColorBalanceType {
     VAProcColorBalanceCount
 } VAProcColorBalanceType;
 
-/** \brief Color standard types. */
+/** Color range types. */
+typedef enum _VAProcColorRange {
+    /** Unknown/unspecified range.
+     *
+     * This will use limited range for YUV and full range for RGB.
+     */
+    VAProcColorRangeUnknown = 0,
+    /** Limited range.
+     *
+     * Used by most video YUV standards, and for RGB in some video contexts.
+     */
+    VAProcColorRangeLimited,
+    /** Full range.
+     *
+     * Used for JPEG images in YUV, and for RGB in all non-video contexts.
+     */
+    VAProcColorRangeFull,
+} VAProcColorRange;
+
+/** Chroma sample location types.
+ *
+ * Only used for YUV 4:2:0 subsampled images.  These values correspond to
+ * the H.264/H.265 syntax element chroma_sample_loc_type_top_field.
+ *
+ */
+typedef enum _VAProcChromaSampleLocation {
+    /** Left.
+     *
+     * This is the default for most video standards (MPEG-2, H.264, H.265),
+     * but typically these can also support other locations.
+     */
+    VAProcChromaSampleLocationLeft = 0,
+    /** Center.
+     *
+     * This is the only option for JPEG and some older video standards (MPEG-1,
+     * H.263).
+     */
+    VAProcChromaSampleLocationCenter = 1,
+    /** Top-left. */
+    VAProcChromaSampleLocationTopLeft = 2,
+    /** Top. */
+    VAProcChromaSampleLocationTop = 3,
+    /** Bottom-left. */
+    VAProcChromaSampleLocationBottomLeft = 4,
+    /** Bottom. */
+    VAProcChromaSampleLocationBottom = 5,
+} VAProcChromaSampleLocation;
+
+/** Color properties. */
+typedef struct _VAProcColorProperties {
+    /** Chroma sample location. */
+    VAProcColorRange color_range;
+    /** Chroma sample location. */
+    VAProcChromaSampleLocation chroma_sample_location;
+    /** Colour primaries, as defined by ISO/IEC 23001-8 and H.273. */
+    uint8_t colour_primaries;
+    /** Transfer characteristics, as defined by ISO/IEC 23001-8 and H.273. */
+    uint8_t transfer_characteristics;
+    /** Matrix coefficients, as defined by ISO/IEC 23001-8 and H.273. */
+    uint8_t matrix_coefficients;
+} VAProcColorProperties;
+
+/** \brief Color standard types.
+ *
+ * These define a set of color properties corresponding to particular
+ * video standards.
+ *
+ * Where matrix_coefficients is specified, it applies only to YUV data -
+ * RGB data always use the identity matrix (matrix_coefficients = 0).
+ */
 typedef enum _VAProcColorStandardType {
     VAProcColorStandardNone = 0,
-    /** \brief ITU-R BT.601. */
+    /** \brief Explicitly specified color properties.
+     *
+     * Use corresponding color properties section.
+     */
+    VAProcColorStandardExplicit,
```

Comment:
Could you move VAProcColorStandardExplicit to the end of union? Otherwise an application built against an old version of libva won't work as expected when VAProcColorStandardBT601 is used.

Reply:
Will do.  (A leftover from the unstable ABI.  The structure size change which was also missing isn't needed, because that will now be inherited from #71.)

Reply:
@FurongZhang  Set the appropriate ```color_standard``` field to ```VAProcColorStandardExplicit``` to indicate that the explicit fields are used, then set the fields in the matching ```color_properties``` structure to your values - ```colour_primaries``` to 14 and ```transfer_characteristics``` to 16.  The ```matrix_coefficients``` field will also need to be set in this case - for RGB it will be 0, while for YUV I believe it uses the BT.2020 NCL value which is 9.

Reply:
@fhvwy Then, I agree with you on this. Is it possible for you to merge this PR to VAAPI? We could use this for HDR color.